### PR TITLE
[Merged by Bors] - TY-2419 Added toDocumentListWithData(), and updated Engine.

### PR DESCRIPTION
**Pull request by rustonaut**
**References:**

- [TY-2419](https://xainag.atlassian.net/browse/TY-2419)

### CODE REVIEW

File: discovery_engine/lib/src/ffi/types/document/document_vec.dart [hunk 1]
```diff
@@ -69,8 +69,13 @@ extension DocumentSliceFfi on List<DocumentFfi> {
     return res;
   }
 
-  List<Document> toDocumentList() => asMap()
+  List<DocumentWithActiveData> toDocumentListWithData() => asMap()
```
Comment (janpetschexain) [inline discovery_engine/lib/src/ffi/types/document/document_vec.dart:72]: maybe for completeness. or maybe we could use a shorthand like `toActiveDocuments()` or sth similar, if this is not already used in other contexts.
```suggestion
  List<DocumentWithActiveData> toDocumentListWithActiveData() => asMap()
```

**Review by janpetschexain — APPROVED** [review]
lgtm

**Comment by rustonaut** [comment]
bors merge

**Comment by rustonaut** [comment]
bors merge

**Comment by rustonaut** [comment]
uh bors failed because the iOS job was spuriously canceled :cry:

**Comment by rustonaut** [comment]
bors merge

**Pull request closed without merging**